### PR TITLE
ci: Refactor linting workflow (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR updates our linting setup to:
1. move the actual linting job into a reusable workflow (like we do for unit tests)
2. split the linting task to run in 3 separate groups `backend`, `frontend`, and `nodes`. 

image

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

File: .github/workflows/linting-reusable.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Reusable linting workflow
```
Comment (ivov) [inline .github/workflows/linting-reusable.yml:1]: I wish there was a way to hide reusables from the GH UI.

**Review by ivov — DISMISSED** [review]

File: .github/workflows/linting-reusable.yml (hunk 1 above)
Comment (netroy) [reply .github/workflows/linting-reusable.yml:1]: for that we should create a `n8n-io/actions` repo, and start moving all re-usable actions/workflows in there.

**Review by despairblue — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Pull request merged**